src/response.js

```diff
@@ -47,4 +47,16 @@ export class Response {
   get text () {
     return this.response.text()
   }
+
+  get isTurboStream () {
+    return this.contentType.match(/^text\/vnd\.turbo-stream\.html/)
+  }
+
+  async applyTurboStream () {
+    if (this.isTurboStream) {
+      document.body.insertAdjacentHTML('beforeend', await this.text)
```

Comment:
@excid3 Just to make sure I understood the idea, doing this would produce the same behavior than doing this ⬇️?
```js
const response = new Request('post', '/myendpoint', {}).perform()
if (response.ok) {
  Turbo.renderStreamMessage(await response.text)
}
```

Reply:
Yeah, it does basically the same thing.

I guess we could refactor this to use the Turbo method if Turbo is defined. Would make things more consistent.

Reply:
Yeah I like the idea, since that in theory if the response of the request is a Turbo Stream probably Turbo will be defined. I'm ok to go through this path and then we can see if someone else has something better in mind.

And @excid3 this will be awesome, actually I'm doing this like I mentioned above, but request.js handle this will be much better ❤️

Reply:
Since `Turbo` isn't necessarily registered on the `window`, we'd have to add an optional dependency on Turbo to use `renderStreamMessage`.

Know of any nice ways to do handle that?

Reply:
I suppose that would require something like this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports it would look like this
```js
let Turbo = window.Turbo
if (Turbo == undefined) {
  Turbo = await import('@hotwired/turbo')
}

if (Turbo) {
  Turbo.renderStreamMessage(await this.text)
} else {
  console.log('log some warning maybe?')
}
```

But this will add a direct dependency with Turbo and I'm not sure if we should do that or not 🤔 

I see some options here that we can choose to not have this dependency

1. Log a warning if `Turbo` isn't registered on the `window` and add a section in the README explaining that if `window.Turbo` is defined the magic will happen.
2. Instead of logging a warning we fallback to the actual implementation.

I'm more inclined to the first one, because if you are receiving a Turbo Stream response you just left to register `Turbo` on the `window` and if you didn't read the docs you probably will see the warning in the logs, but as you mentioned for the newcomers maybe it would be better we fallback to the actual implementation.

Feel free to choose what do you think is the best, and then if needed we improve this in the future.

Reply:
Given that iOS / Android require `window.Turbo` to be defined, I think we should just log a warning. That will be the simplest implementation and we don't have to worry about the Turbo dependency. 👍